src/odemis/dataio/hdf5.py

```diff
@@ -611,262 +611,106 @@ def _parse_physical_data(pdgroup, da):
-        except (KeyError, IndexError, ValueError):
-            pass
+    return das
 
-        # Our extended metadata
-        try:
-            ds = pdgroup["Baseline"]
-            oft = float(ds[i])
-            state = _h5svi_get_state(ds)
-            if state and state[i] == ST_INVALID:
-                raise ValueError
-            md[model.MD_BASELINE] = oft
-        except (KeyError, IndexError, ValueError):
-            pass
 
-        try:
-            ds = pdgroup["IntegrationTime"]
-            it = float(ds[i])  # s
-            state = _h5svi_get_state(ds)
-            if state and state[i] == ST_INVALID:
-                raise ValueError
-            md[model.MD_EXP_TIME] = it
-        except (KeyError, IndexError, ValueError):
-            pass
+# ("PolePosition": (tuple, model.MD_AR_POLE))
+# get_metadata("PolePosition": (tuple, model.MD_AR_POLE))
 
-        try:
-            ds = pdgroup["RefractiveIndexLensImmersionMedium"]
-            state = _h5svi_get_state(ds)
-            if state and state[i] in (ST_INVALID, ST_DEFAULT):
-                raise ValueError
-            ri = float(ds[i])  # ratio
-            md[model.MD_LENS_RI] = ri
-        except (KeyError, IndexError, ValueError):
-            pass
+def get_metadata(pdgroup, index, md, name, MD_key, default_status=False):
+    """
 
-        try:
-            ds = pdgroup["NumericalAperture"]
-            state = _h5svi_get_state(ds)
-            if state and state[i] in (ST_INVALID, ST_DEFAULT):
-                raise ValueError
-            na = float(ds[i])  # ratio
-            md[model.MD_LENS_NA] = na
-        except (KeyError, IndexError, ValueError):
-            pass
+    """
+    try:
+        ds = pdgroup[name]
 
-        try:
-            ds = pdgroup["AccelerationVoltage"]
-            state = _h5svi_get_state(ds)
-            if state and state[i] in (ST_INVALID, ST_DEFAULT):
-                raise ValueError
-            evolt = float(ds[i])  # V
-            md[model.MD_EBEAM_VOLTAGE] = evolt
-        except (KeyError, IndexError, ValueError):
-            pass
+        state = _h5svi_get_state(ds)
+        if default_status:
+            states = (ST_INVALID, ST_DEFAULT)
+        else:
+            states = ST_INVALID
 
-        try:
-            ds = pdgroup["EmissionCurrent"]
-            state = _h5svi_get_state(ds)
-            if state and state[i] in (ST_INVALID, ST_DEFAULT):
-                raise ValueError
-            ecurrent = float(ds[i])  # A
-            md[model.MD_EBEAM_CURRENT] = ecurrent
-        except (KeyError, IndexError, ValueError):
-            pass
+        if state and state[index] == states:
+            raise ValueError
 
-        try:
-            ds = pdgroup["EmissionCurrentOverTime"]
-            state = _h5svi_get_state(ds)
-            if state and state[i] in (ST_INVALID, ST_DEFAULT):
-                raise ValueError
+        # special case
+        if MD_key is model.MD_EBEAM_CURRENT_TIME:
```

Comment:
Please use "==" for the comparison of `md_key`. Most likely, it will work, as you always pass the very same string/constant/object. However, in case this code is copy/pasted, `is` doesn't always work, for instance in the case of writting the string explicitly (instead of using `model.MD_...`).